packages/react-components/src/select/src/Select.tsx

```diff
@@ -1,115 +1,132 @@
      */
-    defaultOpen: bool,
+    defaultOpen?: boolean;
     /**
      * A controlled selected key.
      */
-    selectedKey: string,
+    selectedKey?: string,
     /**
      * The initial value of `selectedKey` when uncontrolled.
      */
-    defaultSelectedKey: string,
+    defaultSelectedKey?: string,
     /**
      * Temporary text that occupies the select trigger when no value is selected.
      */
-    placeholder: string,
+    placeholder?: string,
     /**
      * Whether or not a user input is required before form submission.
      */
-    required: bool,
+    required?: boolean;
     /**
      * Whether or not the select should display as "valid" or "invalid".
      */
-    validationState: oneOf(["valid", "invalid"]),
+    validationState?: "valid" | "invalid";
     /**
      * Called when the select value change.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @param {string} selectedKey - The new selected key.
      * @returns {void}
      */
-    onSelectionChange: func,
+    onSelectionChange?(event: SyntheticEvent, selectedKey: string): void;
     /**
      * Called when the select open state change.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @param {boolean} isOpen - Indicate if the menu is open.
      * @returns {void}
      */
-    onOpenChange: func,
+    onOpenChange?(event: SyntheticEvent, isOpen: boolean): void;
     /**
      * The style to use.
      */
-    variant: oneOf(["outline", "ghost"]),
+    variant?: "outline" | "ghost";
     /**
      * A trigger icon.
      */
-    icon: element,
+    icon?: ReactElement
     /**
      * The direction the select menu will open relative to the input.
      */
-    direction: oneOf(["bottom", "top"]),
+    direction?: "bottom" | "top";
     /**
      * The horizontal alignment of the select menu relative to the input.
      */
-    align: oneOf(["start", "end"]),
+    align?: "start" | "end";
     /**
      * Whether or not the select should autofocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number;
     /**
      * Whether or not the select take up the width of its container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * Whether or not the select is disabled.
      */
-    disabled: bool,
+    disabled?: boolean;
     /**
      * Whether or not the select menu can flip when it will overflow it's boundary area.
      */
-    allowFlip: bool,
+    allowFlip?: boolean;
     /**
      * Whether or not the selection menu position can change to prevent it from being cut off so that it stays visible within its boundary area.
      */
-    allowPreventOverflow: bool,
+    allowPreventOverflow?: boolean;
     /**
      * z-index of the overlay element.
      */
-    zIndex: number,
+    zIndex?: number;
     /**
      * Additional props to render on the menu of options.
      */
-    menuProps: object,
+    menuProps?: SelectMenuProps;
```

Comment:
It should be OverlayProps and maybe rename to overlayProps?

To make it work with the docs, maybe do your extends tricks?